File: testdata/integration-scenarios/preconditionService.ts
/**
 * @ServiceSpec
 * operationId: "strictGetUser"
 * description: "严格的用户获取，要求特定前置条件"
 * preconditions:
 *   "required_user_id": {"!=": [{"var": "user_id"}, null]}
 *   "valid_user_id_format": {"regex": [{"var": "user_id"}, "^[a-zA-Z0-9]{8,}$"]}
 *   "required_method": {"==": [{"var": "http_method"}, "GET"]}
 *   "has_auth_token": {"!=": [{"var": "auth_token"}, null]}
 * postconditions:
 *   "success_status": {"==": [{"var": "span.status.code"}, "OK"]}
 */
export class PreconditionService {
    async strictGetUser(userId: string): Promise<User> {
        return this.userRepository.findById(userId);
    }
}

File: testdata/integration-scenarios/PreconditionService.java
/**
 * @ServiceSpec
 * operationId: "strictCreateUser"
 * description: "严格的用户创建，要求特定前置条件"
 * preconditions:
 *   "required_email": {"!=": [{"var": "request_email"}, null]}
 *   "valid_email_format": {"regex": [{"var": "request_email"}, "^[\\w\\.-]+@[\\w\\.-]+\\.[a-zA-Z]{2,}$"]}
 *   "required_method": {"==": [{"var": "http_method"}, "POST"]}
 *   "min_password_length": {">=": [{"strlen": [{"var": "request_password"}]}, 8]}
 * postconditions:
 *   "success_status": {"==": [{"var": "span.status.code"}, "OK"]}
 */
public class PreconditionService {
    public User strictCreateUser(CreateUserRequest request) {
        return userRepository.save(new User(request));
    }
}
